lib/omnibus/software.rb

```diff
@@ -581,6 +582,14 @@ def builder
       @builder ||= Builder.new(self)
     end
 
+    def manifest_entry
+      Omnibus::ManifestEntry.new(name, {
+                                   source_type: source_type,
+                                   described_version: version,
+                                   locked_version: fetcher.resolved_version,
+                                   locked_source: source})
```

Comment:
@schisamo I've renamed this to_manifest_entry.  On the other two points, here are my thoughts, would love to hear what you think:

1) Re caching: `to_manifest_entry` should return a manifest entry for the Omnibus::Software in its current state, which would complicate caching.  Namely, if we were to cache this, would we then invalidate the caching anytime the source or version was changed on the Omnibus::Software?  Both sides of this are a bit theoretical at the moment since, in a typical run, this will only be called once since we do cache the entire manifest at the project level.

2) I'd like to think of Omnibus::ManifestEntry as a sort of "Type/Value object" that shouldn't change after creation.  This is why I've set everything we care about at instantiation and then only provided reader methods.  Any thoughts on the Ruby-way of doing this?